### PR TITLE
Save to locaStorage  #4

### PR DESCRIPTION
**Pull request by yznetaX**
add methods to save to localStorage and get from it and timer to save  data every 2 seconds

### CODE REVIEW

File: src/components/automated/Conversation.vue [hunk 1]
```diff
@@ -1,32 +1,47 @@
...
-                chain: tree,
-            }
-        },
-        methods: {
-            handelSave() {
-                console.log("chain:",JSON.stringify(this.chain));
-            }
-        }
-    }
+import tree from "./models";
```
Comment (mhdabdelrhman) [inline src/components/automated/Conversation.vue:11]: when you reformat the file, many changes will appear.
- don't use auto format
- format the code manually if possible
- check your files before commit (using Visual code)

image
- the changes must be your code only.

File: src/components/automated/Conversation.vue [hunk 2]
```diff
@@ -1,32 +1,47 @@
...
+    return {
+      chain: tree,
+    };
+  },
+  methods: {
+    handelSave() {
+      localStorage.setItem("chain", JSON.stringify(this.chain));
+      console.log("chain:", JSON.stringify(this.chain));
+    },
+    hanedelLoadChain() {
```
Comment (mhdabdelrhman) [inline src/components/automated/Conversation.vue:28]: functions name should be or like:
- `loadTreeFromLocalStorage()`
- `saveTreeToLocalStorage()`

File: src/components/automated/Conversation.vue [hunk 3]
```diff
@@ -1,32 +1,47 @@
...
+      var newChain = JSON.stringify(this.chain);
+      newChain = localStorage.getItem("chain");
```
Comment (mhdabdelrhman) [inline src/components/automated/Conversation.vue:30]: I think the code will be like
```javascript
var loadedChain = JSON.parse(localStorage.getItem("chain"));
return loadedChain;
```

File: src/components/automated/Conversation.vue [hunk 4]
```diff
@@ -1,32 +1,47 @@
...
+    timer() {
+      setInterval(() => {
+        this.handelSave();
+      }, 2000);
+    },
+  },
+  async created() {
+    await this.handelSave();
+    console.log("this go second");
+    await this.hanedelLoadChain();
```
Comment (mhdabdelrhman) [inline src/components/automated/Conversation.vue:42]: I think the code will be like
```
this.chain = this.loadTreeFromLocalStorage()
```

**Comment by yznetaX** [comment]
`<template>
    <div class="conversation-container">
        <chain-node v-model="chain"></chain-node>
        <div>
            <span @click="saveTreeToLocalStorage">Save</span>
        </div>
    </div>
</template>

<script>
    import tree from './models'
    import ChainNode from './ChainNode'
    export default {
        name: 'conversation',
        components: {
            'chain-node': ChainNode,
        },
        data() {
            return {
                chain: tree,
            }
        },
        methods: {
            saveTreeToLocalStorage() {
                console.log("chain:",JSON.stringify(this.chain));
                localStorage.setItem("chain", JSON.stringify(this.chain));
            },
            loadTreeFromLocalStorage(){
                var loadedChain = JSON.parse(localStorage.getItem("chain"));
                return loadedChain;
            },
            timer() {
                setInterval(() => {
                    this.saveTreeToLocalStorage();
                }, 2000);
    }
        },
        created() {
            this.timer()
        },
    }
</script>
<style></style>
`
i test this and work right 
please take look 
what you say in 40 to 42 
i get an error
and i test this and work fine
@mhdabdelrhman

**Pull request closed without merging**